src/pybind/mgr/cephadm/serve.py

```diff
@@ -709,7 +795,7 @@ def _update_rgw_endpoints(self, rgw_spec: RGWSpec) -> None:
         else:
             self.mgr.remove_health_warning('CEPHADM_RGW')
 
-    def _apply_service(self, spec: ServiceSpec) -> bool:
+    def _apply_service(self, spec: ServiceSpec) -> Optional[Dict[int, Dict[int, Optional[str]]]]:
```

Comment:
Suggestion: can we have some new Type for the returned result? seems really hard to track/understand what are we returning by reading the signature.